src/components/username/Username.tsx

```diff
@@ -0,0 +1,114 @@
+import { useAppDispatch } from "@hooks";
+import { getScrobbles, getUserInfo } from "api/lastfm";
+import classNames from "classnames/bind";
+import { addScrobbles } from "components/upload/uploadSlice";
+import { Track } from "models/ScrobblePage";
+import { User } from "models/User";
+import { FormEvent, useRef, useState } from "react";
+import styles from './Username.module.scss';
+import { setUsername as setUsernameRedux } from "./userSlice";
+
+const cx = classNames.bind(styles);
+
+function UsernameInput() {
+  const [username, setUsername] = useState('');
+  const [finished, setFinished] = useState(false);
+
+  const setTotalCSV = (csv: Track[]) => {
+    console.log('all', csv)
+  }
+
+  const dispatch = useAppDispatch();
+
+  const handleFormSubmit = (event: FormEvent) => {
+    event.preventDefault();
+
+    getUserInfo(username).then((info: User) => {
+      dispatch(setUsernameRedux(info))
+      getUserLastTracks(username, info)
+    })
+  }
+
+  const getUserLastTracks = (
+    async (user: string, info: User) => {
+      if (finished) {
+        return;
+      }
+      try {
+        const pageSize = 200;
+
+        const pc = await getScrobbles(user, 1, pageSize);
+        if (pc[0]["@attr"]?.nowplaying) {
+          pc.shift();
+        }
+
+        const promises: Promise<Track[]>[] = [];
+        const pages = Math.ceil(info!.playcount! / pageSize);
+        console.log({ pages });
+        for (let i = 0; i < pages; i++) {
+          promises.push(getScrobbles(user, i + 1, pageSize));
+        }
+
+        promises.forEach(promise =>
+          promise.then(tracks => {
+            if (tracks[0]["@attr"]?.nowplaying) {
+              tracks.shift();
+            }
+            dispatch(addScrobbles(tracks));
+          })
+        )
+
+        Promise.all(promises).then((trackArray) => {
+          const totalTracks = [];
+          for (const tracks of trackArray) {
+            if (tracks[0]["@attr"]?.nowplaying) {
+              tracks.shift();
+            }
+            totalTracks.push(...tracks);
+          }
+
+          setTotalCSV(totalTracks);
```

Comment:
This may be unnecessary now that we have the `promises.forEach()`.

Should we add a retry logic in case of failed requests?

Reply:
Removed deprecated code. 
Logic for rejected requests will be added in another PR.